interfaces/builtin/gpio.go

```diff
@@ -88,6 +93,26 @@ func (iface *gpioInterface) AppArmorConnectedPlug(spec *apparmor.Specification,
 		return err
 	}
 	path := fmt.Sprint(gpioSysfsGpioBase, number)
+
+	// Export gpio pin here to avoid having a dependency
+	// between the systemd backend and the apparmor backend.
+	//
+	// We also need to check if the gpio symlink is present, if
+	// not it needs exporting. Attempting to export a gpio again
+	// will cause an error on the Write() call.
+	if _, err := os.Stat(fmt.Sprint(gpioSysfsGpioBase, number)); os.IsNotExist(err) {
```

Comment:
isn't there a potential race with the systemd unit itself (that we still create/need) or other places that could be doing this?  interface backend code is run without the state lock nor we synchronize with the systemd unit

Reply:
Yes, good point!